src/test/ui/resolve/token-error-correct.rs

```diff
@@ -15,11 +15,6 @@ fn main() {
     //~^ NOTE: unclosed delimiter
     //~| NOTE: unclosed delimiter
     //~| ERROR: expected expression, found `;`
-    //~| ERROR: unresolved function `foo`
-    //~| NOTE: no resolution found
-    //~| ERROR: unresolved function `bar`
-    //~| NOTE: no resolution found
-    //~| ERROR: expected one of `)`, `,`, `.`, `<`, `?`
```

Comment:
This change highlights that error recovery in the parser no longer works in some cases (that's the purpose of this test indeed!).

So the effect of this patch is that it disables error recovery for one particular randomly selected function `parse_unspanned_seq`.
Yes, this happens to improve diagnostics for the example from https://github.com/rust-lang/rust/issues/44339, but I don't think this is an acceptable solution.

Reply:
What I would do is to introduce the notion of "similar separators" (e.g. `,` and `.` from https://github.com/rust-lang/rust/issues/44339 which are close to each other on a keyboard) and accept separators "similar" to the desired one in `parse_seq_to_before_tokens` after issuing a non-fatal error.

Reply:
Ah I understand. Yes it makes sense to try harder to consume the rest of the sequence before we return. Okay I will explore this option. That will be a different solution so I'll close this PR. Thanks for the feedback!